mlflow/models/model.py

```diff
@@ -698,3 +711,103 @@ def get_model_info(model_uri: str) -> ModelInfo:
+    fail.
+
+    Note that model registry URIs (i.e. URIs in the form ``models:/``) are not
+    supported, as artifacts in the model registry are intended to be read-only.
+
+    If adding requirements, the function will overwrite any existing requirements
+    that overlap, or else append the new requirements to the existing list.
+
+    If removing requirements, the function will ignore any version specifiers,
+    and remove all the specified package names. Any requirements that are not
+    found in the existing files will be ignored.
+
+    Args:
+        model_uri (str): The location, in URI format, of the MLflow model. For example:
+
+            - ``/Users/me/path/to/local/model``
+            - ``relative/path/to/local/model``
+            - ``s3://my_bucket/path/to/model``
+            - ``runs:/<mlflow_run_id>/run-relative/path/to/model``
+            - ``mlflow-artifacts:/path/to/model``
+
+            For more information about supported URI schemes, see
+            `Referencing Artifacts <https://www.mlflow.org/docs/latest/concepts.html#
+            artifact-locations>`_.
+
+        operation (Literal["add", "remove]): The operation to perform.
+            Must be one of "add" or "remove".
+
+        requirement_list (List[str]): A list of requirements to add or remove from the model.
+            For example: ["numpy==1.20.3", "pandas>=1.3.3"]
+    """
+    if ModelsArtifactRepository.is_models_uri(model_uri):
+        raise MlflowException(
+            f'Failed to set requirements on "{model_uri}". '
+            + "Model URIs with the `models:/` scheme are not supported.",
+            INVALID_PARAMETER_VALUE,
+        )
+
+    resolved_uri = model_uri
+    if RunsArtifactRepository.is_runs_uri(model_uri):
+        resolved_uri = RunsArtifactRepository.get_underlying_uri(model_uri)
+
+    _logger.info(f"Retrieving model requirements files from {resolved_uri}...")
+    local_paths = get_model_requirements_files(resolved_uri)
+    conda_yaml_file = local_paths[_CONDA_ENV_FILE_NAME]
+    requirements_txt_file = local_paths[_REQUIREMENTS_FILE_NAME]
+
+    if operation == "add":
+        _logger.info(f"Adding requirements to {_CONDA_ENV_FILE_NAME}...")
+        _add_requirements_to_file(requirement_list, conda_yaml_file)
+        _logger.info(f"Adding requirements to {_REQUIREMENTS_FILE_NAME}...")
+        _add_requirements_to_file(requirement_list, requirements_txt_file)
+    else:
+        _logger.info(f"Removing requirements from {_CONDA_ENV_FILE_NAME}...")
+        _remove_requirements_from_file(requirement_list, conda_yaml_file)
+        _logger.info(f"Removing requirements from {_REQUIREMENTS_FILE_NAME}...")
+        _remove_requirements_from_file(requirement_list, requirements_txt_file)
+
+    _logger.info(f"Uploading updated requirements files to {resolved_uri}...")
+    _upload_artifact_to_uri(conda_yaml_file, resolved_uri)
+    _upload_artifact_to_uri(requirements_txt_file, resolved_uri)
```

Comment:
Nit: is it possible to download both files using `log_artifacts`?

Reply:
I think the impact would be subtle, but we can save one HTTP request for getting pre-signed URLs.

Reply:
I took a look at this—do you have a code pointer for where the batching occurs? It looks like in some artifact repos, the `log_artifacts()` function just walks across all files in a directory and calls `log_artifact()` on each one:

https://github.com/mlflow/mlflow/blob/dc51fc6b82fa78a30f8be5efed18f7867809311d/mlflow/store/artifact/http_artifact_repo.py#L79-L80

Reply:
Batching occurs when we generate pre-signed URLs on Databricks.

Reply:
https://github.com/mlflow/mlflow/blob/7380c98ae4473a9746c25ef04061eb1c46fbd32e/mlflow/store/artifact/cloud_artifact_repo.py#L89

Reply:
This is NOT a blocker.